lib/code-ribbon-single-ribbon.js

```diff
@@ -50,11 +54,47 @@ class CodeRibbonSingleRibbon extends PaneAxis {
     };
   }
 
+  at_least_n_empty_right_patches(target_num = 2) {
+    if (target_num > this.children.length) return false;
+    for (var i = 1; i <= target_num; i++) {
+      if (
+        this.children[this.children.length -i].getItems().length > 0
+      ) {
+        return false;
+      }
+    }
+    // if there weren't any panes that had at least one item:
+    return true;
+  }
+
+  /**
+   * returns the length of this ribbon not including trailing empty patches
+   * @return {number} the length of real patches
+   */
+  get_contentful_length() {
+    for (var i = this.children.length - 1; i >= 0; i--) {
+      if (
+        this.children[i].getItems().length > 0
+      ) {
+        return i+1;
+      }
+    }
+    return 0;
+  }
+
+  remove_rightmost_patch() {
+    var target_patch = this.children[this.children.length -1];
+    if (target_patch.getItems().length > 0) {
+      crlogger.warn("Removing a Patch which isn't empty: ", target_patch);
+    }
+    target_patch.destroy();
+  }
+
   cr_update() {
-    // add panes to fill at least one screen:
+    // add panes to meet quota:
```

Comment:
what is the defined quota honestly, is it just one column of patches past whatever the user sees in front of them depending on their settings?

Reply:
The quota changes dynamically based on the contents of the ribbons, although all ribbons should meet the same minimum quota so that the element sizes are still correctly balanced.

Right now it's the `contentful_length + patches_per_screen - 1`